### PR TITLE
An inactive user shouldn't be able to be added as a collaborator

### LINKED ISSUE #4534: An inactive user can be added as a collaborator
- Gitea version (or commit ref):  a74426d631e3311cc3c460ae9917f76f0221f4de
- Git version: 2.15.2
- Operating system: MacOS (High Sierra)
- Database (use `[x]`):
  - [ ] PostgreSQL
  - [ ] MySQL
  - [ ] MSSQL
  - [ ] SQLite
- Can you reproduce the bug at https://try.gitea.io:
  - [ ] Yes (provide example URL)
  - [ ] No
  - [x] Not relevant
- Log gist:

## Description

An inactive user can be added as a collaborator in a repository since Inactive means the user hasn't been verified.

### PR DESCRIPTION
**Pull request by adelowo**
Fixes #4534

### CODE REVIEW

**Comment by lafriks** [comment]
This check must be also added in API

File: options/locale/locale_en-US.ini [hunk 1]
```diff
@@ -1014,6 +1014,7 @@ settings.transfer_succeed = The repository has been transferred.
 settings.confirm_delete = Delete Repository
 settings.add_collaborator = Add Collaborator
 settings.add_collaborator_success = The collaborator has been added.
+settings.add_collaborator_inactive_user = Cannot add an inactive user as a collaborator.
```
Comment (lafriks) [inline options/locale/locale_en-US.ini]: `Can not`

**Review by lafriks — APPROVED** [review]
Test would be nice for this

**Comment by adelowo** [comment]
~~I am trying to add some tests but it keeps on failing... would fix tomorrow~~

@lafriks Done

**Comment by lunny** [comment]
LGTM

**Comment by lunny** [comment]
@adelowo  conflicted

**Comment by lafriks** [comment]
Resolved confilcts

**Review by sapk — APPROVED** [review]

**Comment by adelowo** [comment]
Thanks @lafriks

**Comment by mjmlvp** [comment]
So I'd like to pitch in. Though a bit late as this seems to already be a done deal.

I would object to it being impossible to add an inactive user as a colaborator.

An inactive user added as a collaborator, I would interpret as a 'inactive collaborator'. Nothing strange in principle about that. To me the issue would more be, do you list inactive collaborators, or do you maybe only list them to team administrators &owners?

I think not allowing this is an arbitrary restriction in workflow. For example an admin may be in the process of setting up a new user's account, giving him the correct permissions etc., keeping the account under creation hidden/inactive.
Then as a final step after all account details check out correct, the admin may like to activate the account. 

I would be interested to know how others view this.

**Pull request merged**